iothub/device/src/Pipeline/PipelineContext.cs

```diff
@@ -7,14 +7,8 @@ namespace Microsoft.Azure.Devices.Client
 {
     internal class PipelineContext
     {
-        internal IotHubClientTransportSettings TransportSettings { get; set; }
```

Comment:
PipelineContext holds IotHubConnectionInfo, which in turn holds ClientOptions. TransportSettings and ProductInfo are a part of ClientOptions. So, we don't need PipelineContext to hold them explicitly.